Microsoft.Toolkit.Uwp.UI.Controls.Layout/GridSplitter/SplitBase.cs

```diff
@@ -0,0 +1,220 @@
+            set { SetValue(GripperForegroundProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="GripperForeground"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty GripperForegroundProperty =
+            DependencyProperty.Register(nameof(GripperForeground), typeof(Brush), typeof(SplitBase), new PropertyMetadata(default(Brush)));
+
+        /// <summary>
+        /// Gets or sets the direction that the sizer will interact with.
+        /// </summary>
+        public ContentResizeDirection ResizeDirection
+        {
+            get { return (ContentResizeDirection)GetValue(ResizeDirectionProperty); }
+            set { SetValue(ResizeDirectionProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="ResizeDirection"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty ResizeDirectionProperty =
+            DependencyProperty.Register(nameof(ResizeDirection), typeof(ContentResizeDirection), typeof(SplitBase), new PropertyMetadata(ContentResizeDirection.Vertical));
+
+        /// <summary>
+        /// Gets or sets the control that the <see cref="SplitBase"/> is resizing. Be default, this will be the visual ancestor of the <see cref="SplitBase"/>.
+        /// </summary>
+        public FrameworkElement TargetControl
+        {
+            get { return (FrameworkElement)GetValue(TargetControlProperty); }
+            set { SetValue(TargetControlProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="TargetControl"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty TargetControlProperty =
+            DependencyProperty.Register(nameof(TargetControl), typeof(FrameworkElement), typeof(SplitBase), new PropertyMetadata(null, OnTargetControlChanged));
+
+        private static void OnTargetControlChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            // Check if our width can be manipulated
+            if (d is SplitBase splitterBase && e.NewValue is FrameworkElement element)
+            {
+                // TODO: For Auto we might want to do detection logic (TBD) here first?
+                if (splitterBase.ResizeDirection != ContentResizeDirection.Horizontal && double.IsNaN(element.Width))
+                {
+                    element.Width = element.DesiredSize.Width;
+                }
+
+                if (splitterBase.ResizeDirection != ContentResizeDirection.Vertical && double.IsNaN(element.Height))
+                {
+                    element.Height = element.DesiredSize.Height;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the <see cref="SplitBase"/> control is resizing in the opposite direction.
+        /// </summary>
+        public bool InvertDragDirection
```

Comment:
`IsDragInverted` maybe as booleans typically start with `Is`?

Also this property is only useful for ContentSizer right, so we should put it there vs. the base class?

Reply:
Unless we're expecting devs to inherit from `ContentSizer`, and not `SplitBase`, in which case that renders a few of my talking points moot.